projects/addon-table/components/table-pagination/table-pagination.component.ts

```diff
@@ -108,15 +139,83 @@ export class TuiTablePaginationComponent {
         };
     }
 
+    /**
+     * @deprecated Use `selectPreviousPage()` instead.
+     */
     protected back(): void {
-        this.page--;
-        this.pageChange.emit(this.page);
-        this.paginationChange.emit(this.pagination);
+        this.selectPreviousPage();
     }
 
+    /**
+     * @deprecated Use `selectNextPage()` instead.
+     */
     protected forth(): void {
-        this.page++;
-        this.pageChange.emit(this.page);
+        this.selectNextPage();
+    }
+
+    protected selectFirstPage(): void {
+        this.selectPage(this.firstPage);
+    }
+
+    protected selectPreviousPage(): void {
+        const previousPage = this.previousPage;
+
+        if (previousPage === null) {
+            return;
+        }
+
+        this.selectPage(previousPage);
+    }
+
+    protected selectNextPage(): void {
+        const nextPage = this.nextPage;
+
+        if (nextPage === null) {
+            return;
+        }
+
+        this.selectPage(nextPage);
+    }
+
+    protected selectLastPage(): void {
+        this.selectPage(this.lastPage);
+    }
+
+    /**
+     * Returns the index of the first page.
+     */
+    private get firstPage(): number {
```

Comment:
This getter makes no sense. It's obvious that first index is 0, there's no need to move into a semantic getter.